### PR TITLE
1.5.1 update

### PR DESCRIPTION
**Pull request by cbouss**
Update pandas to 1.5.1

Changes:
- Update version
- Update dependencies
- Add run_constrained for optional dependencies
- Update about section
- Disable parallel build

Links:
home: https://pandas.pydata.org
dev: https://github.com/pandas-dev/pandas/tree/v1.5.1
release note: https://pandas.pydata.org/pandas-docs/version/1.5.1/whatsnew/index.html
backward incompatible changes: https://pandas.pydata.org/pandas-docs/version/1.5.1/whatsnew/v1.5.0.html#backwards-incompatible-api-changes
installation doc: https://pandas.pydata.org/pandas-docs/version/1.5.1/getting_started/install.html#installation
setup: https://github.com/pandas-dev/pandas/blob/v1.5.1/setup.cfg

Tests:
Tested downstreams packages. One regression found with pandas-profiling.
Pandas-profiling update: https://github.com/AnacondaRecipes/pandas-profiling-feedstock/pull/7

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,25 +25,57 @@ requirements:
...
-    - numexpr >=2.7.1
-    - bottleneck >=1.3.1
+    # see https://pandas.pydata.org/pandas-docs/version/1.5.1/getting_started/install.html#recommended-dependencies
+    - numexpr >=2.7.3
+    - bottleneck >=1.3.2
+  run_constrained:
+    # Optional dependencies
+    # see https://pandas.pydata.org/pandas-docs/version/1.5.1/getting_started/install.html#optional-dependencies
+    - tzdata >=2022a
+    - matplotlib >=3.3.2
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:50]: `matplotlib-base` would be better

File: recipe/meta.yaml [hunk 2]
```diff
@@ -24,25 +25,57 @@ requirements:
...
+    - pyxlsb >=1.0.8
+    - beautifulsoup4 >=4.9.3
+    - html5lib >=1.1
+    - lxml >=4.6.3
+    - sqlalchemy >=1.4.16
+    - psycopg2 >=2.8.6
+    - pymysql >=1.0.2
+    - pytables >=3.6.1
+    - blosc >=1.21.0
+    - fastparquet >=0.4.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:69]: missing `zlib` https://pandas.pydata.org/pandas-docs/version/1.5.1/getting_started/install.html#other-data-sources ?

Comment (cbouss) [reply recipe/meta.yaml:69]: I did not list it because there was no version constraint on zlib.

**Comment by cbouss** [comment]
> https://github.com/pandas-dev/pandas/blob/v1.5.1/environment.yml#L25

There are some discrepancies between environment.yml, setup.py and the release note.
I prefer to rely on setup.py and the release note.

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: addressed in commit, thank you

**Comment by chenghlee** [comment]
> There are some discrepancies between environment.yml, setup.py and the release note.
I prefer to rely on setup.py and the release note.

That's the correct path. The upstream's `environment.yml` is for setting up a pandas development environment, so we should not consider it when deciding what to include as run requirements or run constraints in our package.

**Review by skupr-anaconda — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**